package/endpoint_adapter/src/endpoint_adapter.c

```diff
@@ -698,287 +766,256 @@ static ssize_t handle_write_all_via_framer(handle_t *handle,
 {
-  if (nonblock && write_fd != -1) {
+  if (write_fd != -1) {
     int arg = O_NONBLOCK;
     fcntl(write_fd, F_SETFL, &arg);
   }
-  switch (endpoint_mode) {
-  case ENDPOINT_PUBSUB: {
-    if (pub_addr != NULL && read_fd != -1) {
-      debug_printf("Forking for pub\n");
-      pid_t pid = fork();
-      if (pid == 0) {
-        setup_metrics("pub");
-        /* child process */
-        pk_endpoint_t *pub = pk_endpoint_start(PK_ENDPOINT_PUB);
-        if (pub == NULL) {
-          debug_printf("pk_endpoint_start(PK_ENDPOINT_PUB) returned NULL\n");
-          exit(EXIT_FAILURE);
-        }
 
-        /* Read from fd, write to pub */
-        handle_t pub_handle;
-        if (handle_init(&pub_handle,
-                        pub,
-                        -1,
-                        -1,
-                        framer_name,
-                        filter_in_name,
-                        filter_in_config,
-                        false)
-            != 0) {
-          debug_printf("handle_init for pub returned error\n");
-          exit(EXIT_FAILURE);
-        }
-
-        handle_t fd_handle;
-        if (handle_init(&fd_handle,
-                        NULL,
-                        read_fd,
-                        -1,
-                        FRAMER_NONE_NAME,
-                        FILTER_NONE_NAME,
-                        NULL,
-                        is_can)
-            != 0) {
-          debug_printf("handle_init for read_fd returned error\n");
-          exit(EXIT_FAILURE);
-        }
+  if (read_fd != -1) {
+    int arg = O_NONBLOCK;
+    fcntl(write_fd, F_SETFL, &arg);
+  }
 
-        io_loop_pubsub(&fd_handle, &pub_handle);
-        pk_endpoint_destroy(&pub);
-        assert(pub == NULL);
-        handle_deinit(&pub_handle);
-        handle_deinit(&fd_handle);
-        pk_metrics_destroy(&MR);
-        debug_printf("Exiting from pub fork\n");
-        exit(EXIT_SUCCESS);
-      } else {
-        /* parent process */
-        pub_pid = pid;
-      }
+  if (fork_needed) {
+    pid_t pid = fork();
+    if (pid != 0) {
+      return 0;
     }
+  }
 
-    if (sub_addr != NULL && write_fd != -1) {
-      debug_printf("Forking for sub\n");
-      pid_t pid = fork();
-      if (pid == 0) {
-        setup_metrics("sub");
-        /* child process */
-        pk_endpoint_t *sub = pk_endpoint_start(PK_ENDPOINT_SUB);
-        if (sub == NULL) {
-          debug_printf("pk_endpoint_start(PK_ENDPOINT_SUB) returned NULL\n");
-          exit(EXIT_FAILURE);
-        }
+  loop_ctx.loop = pk_loop_create();
+  setup_metrics();
 
-        /* Read from sub, write to fd */
-        handle_t sub_handle;
-        if (handle_init(&sub_handle, sub, -1, -1, FRAMER_NONE_NAME, FILTER_NONE_NAME, NULL, false)
-            != 0) {
-          debug_printf("handle_init for sub returned error\n");
-          exit(EXIT_FAILURE);
-        }
+  void *handle = pk_loop_timer_add(loop_ctx.loop, 1000, do_metrics_flush, NULL);
```

Comment:
Is this unused variable or don't I just see it..

Reply:
Only purpose is for the assert on the next line